core/src/main/java/hudson/PluginFirstClassLoader2.java

```diff
@@ -0,0 +1,167 @@
+     *       <p>Invoke {@link #loadClass(String)} on the parent class loader.
+     * </ol>
+     *
+     * <p>If the class was found using the above steps and the {@code resolve} flag is true, this
+     * method will then invoke {@link #resolveClass(Class)} on the resulting {@link Class} object.
+     *
+     * <p>This method synchronizes on the result of {@link #getClassLoadingLock(String)} during the
+     * entire class loading process.
+     *
+     * @param name The binary name of the class
+     * @param resolve If {@code true} then resolve the class
+     * @return The resulting {@link Class} object
+     * @throws ClassNotFoundException If the class could not be found
+     */
+    @Override
+    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        synchronized (getClassLoadingLock(name)) {
+            Class<?> c = findLoadedClass(name);
+            if (c == null && systemClassLoader != null) {
+                try {
+                    c = systemClassLoader.loadClass(name);
+                } catch (ClassNotFoundException e) {
+                    // ignore
+                }
+            }
+            if (c == null) {
+                try {
+                    c = findClass(name);
+                } catch (ClassNotFoundException e) {
+                    // ignore
+                }
+            }
+            if (c == null) {
+                c = getParent().loadClass(name);
+            }
+            if (resolve) {
+                resolveClass(c);
+            }
+            return c;
+        }
+    }
+
+    /**
+     * Find the resource with the given name. A resource is some data (images, audio, text, etc)
+     * that can be accessed by class code in a way that is independent of the location of the code.
+     *
+     * <p>The name of a resource is a '{@code /}'-separated path name that identifies the resource.
+     * This method searches for resources in the following order:
+     *
+     * <ol>
+     *   <li>
+     *       <p>Invoke {@link #getResource(String)} on the system class loader.
+     *   <li>
+     *       <p>Invoke {@link #findResource(String)} to find the resource.
+     *   <li>
+     *       <p>Invoke {@link #getResource(String)} on the parent class loader.
+     * </ol>
+     *
+     * @param name The resource name
+     * @return {@link URL} object for reading the resource; {@code null} if the resource could not
+     *     be found, a {@link URL} could not be constructed to locate the resource, the resource is
+     *     in a package that is not opened unconditionally, or access to the resource is denied by
+     *     the security manager.
+     * @throws NullPointerException If {@code name} is {@code null}
+     */
+    @Override
+    public URL getResource(String name) {
+        Objects.requireNonNull(name);
+        URL url;
+        if (systemClassLoader != null) {
+            url = systemClassLoader.getResource(name);
+        } else {
```

Comment:
similar to previous - I think this would have been consulted as part of the parent before so should come after looking for resources in this plugin?

Reply:
I removed any references to the system class loader.